app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -465,13 +465,15 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     )
 
     /**
-     * Caches the last known "is default browser" state when the app was paused.
-     * For an up to do date state use `isDefaultBrowser` instead.
+     * Declared as a function for performance purposes. When this class is first initialized, the
+     * isDefaultBrowser function takes approx. 30-40ms.
      */
-    var wasDefaultBrowserOnLastResume by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_default_browser),
-        default = isDefaultBrowser()
-    )
+    fun checkDefaultBrowserAndSet(): Boolean {
+        val prefKey = appContext.getPreferenceKey(R.string.pref_key_default_browser)
+        val isDefaultBrowser = isDefaultBrowser()
+        this.preferences.edit().putBoolean(prefKey, isDefaultBrowser).apply()
+        return this.preferences.getBoolean(prefKey, isDefaultBrowser) != isDefaultBrowser
```

Comment:
I think I may be misunderstanding this. On 474 we are setting prefKey value to = `isDefaultBrowser`, and then on the next line we are checking if the prefKey value = `isDefaultBrowser`... I don't see how the != could ever fail 🤔

Reply:
I think you're right.  I was trying to mimic the behavior from `HomeActivity` since I didn't want to use the sharedPreferences in there... I kinda inverted the behavior by accident (where `HomeActivity` does the check and then it sets it). Thank you for noticing that!